### PR TITLE
save actual Nokogiri UTF-8 chars

### PR DESCRIPTION
**Pull request by fraudfilter**
Nokorigi applies the encoding settings for the nokogiri-object through "encoding" attribute by default, please save the actual characters that reverts nokogiri for UTF-8!

### CODE REVIEW

**Comment by stevecheckoway** [comment]
I don't understand what you mean by "save the actual characters that reverts nokogiri for UTF-8." The patch simply sets the encoding to to UTF-8 but that doesn't appear to actually do anything to the underlying xmlDoc (see the source for [Nokogiri::XML::Document#encoding=](http://www.rubydoc.info/github/sparklemotion/nokogiri/Nokogiri%2FXML%2FDocument:encoding=)).

If all you want is to record the fact that the encoding is UTF-8, that seems reasonable, but I can't tell from your comment if you want something else.

File: test-nokogumbo.rb [hunk 1]
```diff
@@ -53,27 +54,27 @@ def test_nil
     def test_macroman_encoding
       mac="<span>\xCA</span>".force_encoding('macroman')
       doc = Nokogiri::HTML5(mac)
-      assert_equal '<span>&#xA0;</span>', doc.at('span').to_xml
+      assert_equal "<span> </span>", doc.at("span").to_xml
```
Comment (stevecheckoway) [inline test-nokogumbo.rb]: This change makes it difficult to see what character is expected. It's also not related to  testing your actual change.

File: test-nokogumbo.rb [hunk 2]
```diff
@@ -53,27 +54,27 @@ def test_nil
...
     end
 
     def test_iso8859_encoding
       iso8859="<span>Se\xF1or</span>".force_encoding(Encoding::ASCII_8BIT)
       doc = Nokogiri::HTML5(iso8859)
-      assert_equal '<span>Se&#xF1;or</span>', doc.at('span').to_xml
+      assert_equal '<span>Señor</span>', doc.at('span').to_xml
```
Comment (stevecheckoway) [inline test-nokogumbo.rb]: This is also not testing your change.

File: test-nokogumbo.rb [hunk 3]
```diff
@@ -53,27 +54,27 @@ def test_nil
...
     end
 
     def test_charset_encoding
       utf8="<meta charset='utf-8'><span>Se\xC3\xB1or</span>".
         force_encoding(Encoding::ASCII_8BIT)
       doc = Nokogiri::HTML5(utf8)
-      assert_equal '<span>Se&#xF1;or</span>', doc.at('span').to_xml
+      assert_equal '<span>Señor</span>', doc.at('span').to_xml
```
Comment (stevecheckoway) [inline test-nokogumbo.rb]: Same here.

File: test-nokogumbo.rb [hunk 4]
```diff
@@ -53,27 +54,27 @@ def test_nil
...
     end
 
     def test_bogus_encoding
       bogus="<meta charset='bogus'><span>Se\xF1or</span>".
         force_encoding(Encoding::ASCII_8BIT)
       doc = Nokogiri::HTML5(bogus)
-      assert_equal '<span>Se&#xF1;or</span>', doc.at('span').to_xml
+      assert_equal '<span>Señor</span>', doc.at('span').to_xml
```
Comment (stevecheckoway) [inline test-nokogumbo.rb]: And here.

File: test-nokogumbo.rb [hunk 5]
```diff
@@ -137,6 +138,22 @@ def test_parse_fragment_errors
...
+        </head>
+        <body>
+          Кирилические символы
+        </body>
+      </html>
+    TEXT
+    doc = Nokogiri::HTML5.parse(html)
+    assert_equal "UTF-8", doc.encoding
+    assert_equal "Кирилические символы", doc.at('body').text.gsub(/\n\s+/,'')
+  end
```
Comment (stevecheckoway) [inline test-nokogumbo.rb]: The second assertion isn't related to the encoding. I think the test should be something like

``` ruby
def test_document_encoding
  doc = Nokogiri::HTML5.parse(buffer)
  assert_equal 'UTF-8', doc.encoding
```

**Comment by fraudfilter** [comment]
I want the UTF-8 symbols at the output be the same as they were at the input (like using Nokogiri).

Unfortunately, I'm not a native English speaker, that's why I can express my thoughts not clear enough. So I attached the screenhots to help to explain what I mean.

**Comment by stevecheckoway** [comment]
Thank you. Now I understand what's going on. This seems reasonable to me.

**Comment by rubys** [comment]
No objections here.  Given that it changes the results of a test, I'd be inclined to bump the version number to 1.5.0 before releasing it.

**Comment by stevecheckoway** [comment]
Thanks for this @fraudfilter.

**Pull request merged**